### PR TITLE
Rename PresetsRepository

### PR DESCRIPTION
**Pull request by PaulKlauser**
The name already didn't match the responsibility of the class, and with the migration, the name was even more confusing.

#452

### CODE REVIEW

File: app/src/main/java/com/willowtree/vocable/presets/ILegacyCategoriesAndPhrasesRepository.kt [hunk 1]
```diff
@@ -6,8 +6,9 @@ import com.willowtree.vocable.room.PhraseDto
 import com.willowtree.vocable.utils.locale.LocalesWithText
 import kotlinx.coroutines.flow.Flow
 
-//TODO: PK - Rename this once we make the jump to rename [PresetsRepository] -> "RoomPresetsRepository"
-interface IPresetsRepository {
+@Deprecated("This is the old way of accessing categories and phrases. Prefer using" +
+        " ICategoriesUseCase instead.")
```
Comment (mattttvaughn) [inline app/src/main/java/com/willowtree/vocable/presets/ILegacyCategoriesAndPhrasesRepository.kt:10]: Nit but would we want `PhrasesUseCase` mentioned in here as well?

**Review by mattttvaughn — APPROVED** [review]
Name change here is great 👍

Is the implication that we'll also eventually deprecate `StoredCategoriesRepository` and prefer exposing everything through the UseCases? Or was this guy just too insanely overloaded with stuff?

File: app/src/main/java/com/willowtree/vocable/presets/ILegacyCategoriesAndPhrasesRepository.kt (hunk 1 above)
Comment (PaulKlauser) [reply app/src/main/java/com/willowtree/vocable/presets/ILegacyCategoriesAndPhrasesRepository.kt:10]: Good call, I forgot that existed

**Comment by PaulKlauser** [comment]
> Name change here is great 👍
> 
> Is the implication that we'll also eventually deprecate `StoredCategoriesRepository` and prefer exposing everything through the UseCases? Or was this guy just too insanely overloaded with stuff?

`StoredCategoriesRepository` will still exist, but the UI will only interact via the UseCases, which will then delegate to the appropriate stored/preset repositories as necessary

File: app/src/main/java/com/willowtree/vocable/presets/ILegacyCategoriesAndPhrasesRepository.kt (hunk 1 above)
Comment (PaulKlauser) [reply app/src/main/java/com/willowtree/vocable/presets/ILegacyCategoriesAndPhrasesRepository.kt:10]: I'ma fix this forward in a following PR 👍

**Pull request merged**